app/components/primer/component.rb

```diff
@@ -5,5 +5,31 @@ class Component < ViewComponent::Base
     include ClassNameHelper
     include FetchOrFallbackHelper
     include OcticonsHelper
+
+    class << self
+      ##
+      # Creates a `deprecations` class attribute for each child
+      # This will ensure deprecations between different components
+      # won't clash
+      def inherited(base)
+        base.class_eval do
+          class_attribute :deprecations, instance_writer: false
+          self.deprecations = {}
+        end
+
+        super
+      end
+
+      def deprecated_options_for(opt, *args)
+        deprecations[opt] = args
+      end
+    end
+
+    def before_render
+      deprecations.each do |(opt, values)|
+        val = instance_variable_get("@#{opt}")
```

Comment:
What happens if we never assign the value to an instance variable, or if we do, if such assignment is removed?

Reply:
then `val` will be `nil` and not warn :/

I couldn't find any good (non-intrusive) way of checking the values

Reply:
Hmm. What if we were to pass the deprecations to `Classify` as part of `kwargs`?

Reply:
hm, but not all deprecated options are from classify 🤔

Reply:
Seeing how this is only for our components, I think we can be pretty strict and raise an error. I assume we'd address the errors in the same PR.

Reply:
done 😉